src/index.js

```diff
@@ -4,15 +4,21 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { CategoryProvider } from "./Context/CategoryContext";
+import { BrowserRouter } from "react-router-dom";
+import { ProductProvider } from "./Context/ProductContext";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <CategoryProvider>
-      <App />
-    </CategoryProvider>
+    <ProductProvider>
+      <CategoryProvider>
+        <BrowserRouter>
```

Comment:
we have to provide browser router in the top level

Reply:
Ok ..done..thanks for the suggestion.